### PR TITLE
feat: refactor User Settings

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes https://linear.app/nango/issue/NAN-1430/refacto-profile-settings

### Major changes
- New user settings page
- Add related endpoints

### Minor changes
- Reduce session duration
I just happen to notice that long session period. This is debatable but all security recommendations are stricter than that, and we store information there that can be outdated + we show credentials, it seems like it's a good idea to go lower (ideally max 24h but it's also very annoying)

- Remove unused types
- Renamed WebUser to ApiUser for consistency

### CODE REVIEW

File: packages/shared/lib/services/user.service.ts [hunk 1]
```diff
@@ -9,17 +9,9 @@ const VERIFICATION_EMAIL_EXPIRATION = 3 * 24 * 60 * 60 * 1000;
...
-        if (result == null || result.length == 0 || result[0] == null) {
-            return null;
-        }
-
-        if (result[0].suspended) {
-            return null;
-        }
-
-        return result[0];
+        return result && !result.suspended ? result : null;
```
Comment (TBonnin) [inline packages/shared/lib/services/user.service.ts]: could the `suspended` be part of the query?

File: packages/shared/lib/services/user.service.ts [hunk 2]
```diff
@@ -91,23 +83,15 @@ class UserService {
     }
 
     async getUserByEmail(email: string): Promise<User | null> {
-        const result = await db.knex.select('*').from<User>(`_nango_users`).where({ email: email });
-
-        if (result == null || result.length == 0 || result[0] == null) {
-            return null;
-        }
+        const result = await db.knex.select('*').from<User>(`_nango_users`).where({ email: email }).first();
```
Comment (TBonnin) [inline packages/shared/lib/services/user.service.ts:82]: we don't check `suspended` in this function and the one below? Is it intended?

File: packages/shared/lib/services/user.service.ts [hunk 3]
```diff
@@ -169,11 +149,13 @@ class UserService {
...
-    async update({ id, ...data }: { id: number } & Omit<Partial<DBUser>, 'id'>): Promise<number> {
-        return await db.knex
+    async update({ id, ...data }: { id: number } & Omit<Partial<DBUser>, 'id'>): Promise<DBUser | null> {
+        const up = await db.knex
             .from<DBUser>(`_nango_users`)
             .update({ ...data, updated_at: new Date() })
-            .where({ id });
+            .where({ id })
+            .returning('*');
+        return up[0] || null;
```
Comment (TBonnin) [inline packages/shared/lib/services/user.service.ts]: you can also do 
```
const [up] = await ...
return up || null
````

File: packages/webapp/src/components/PrivateRoute.tsx [hunk 4]
```diff
@@ -71,7 +71,7 @@ export const PrivateRoute: React.FC = () => {
         }
     }, [user, identify]);
 
-    if (loading || !ready) {
+    if (loading || !ready || loadingUser) {
```
Comment (TBonnin) [inline packages/webapp/src/components/PrivateRoute.tsx]: what's the difference between loading and loadingUser

File: packages/server/lib/clients/auth.client.ts [hunk 5]
```diff
@@ -32,7 +33,7 @@ export function setupAuth(app: express.Router) {
             store: sessionStore,
             name: 'nango_session',
             unset: 'destroy',
-            cookie: { maxAge: 30 * 24 * 60 * 60 * 1000, secure: false },
+            cookie: { maxAge: 7 * 24 * 60 * 60 * 1000, secure: false },
```
Comment (TBonnin) [inline packages/server/lib/clients/auth.client.ts:36]: at some point we are gonna need to review our cookie policy. In the meantime 7 days is probably a good compromise

**Review by TBonnin — APPROVED** [review]

File: packages/webapp/src/components/PrivateRoute.tsx (hunk 4 above)
Comment (bodinsamuel) [reply packages/webapp/src/components/PrivateRoute.tsx]: let me rename the other loading 👌🏻

File: packages/shared/lib/services/user.service.ts (hunk 1 above)
Comment (bodinsamuel) [reply packages/shared/lib/services/user.service.ts]: true

File: packages/shared/lib/services/user.service.ts (hunk 2 above)
Comment (bodinsamuel) [reply packages/shared/lib/services/user.service.ts:82]: good point

File: packages/server/lib/clients/auth.client.ts (hunk 5 above)
Comment (bodinsamuel) [reply packages/server/lib/clients/auth.client.ts:36]: yes, long sessions are a liability

**Comment by bodinsamuel** [comment]
Thanks for the review @TBonnin it forced me to retest and realized I messed up a condition just before pushing 👌🏻

**Pull request merged**